sal: fix stale BLE advert state on deinit and reconnect (#789)
nrfconnect/sdk-sidewalk: doc/known_issues.rst

  **Affected platforms:** The nRF52840 SoC and the nRF54L15 SoC with the LR1110 shield

> KRKNWK-22208: BLE advertising fails after Sidewalk deinit and reinit cycle

Reviewer: ```suggestion
KRKNWK-22208: Bluetooth LE advertising fails after Sidewalk deinit and reinit cycle
```

  After a full Sidewalk BLE deinitialization and reinitialization cycle, advertising may fail to restart.

>   When a BLE connection is established, the Bluetooth stack stops advertising, but the internal advertising state may remain stale.

Reviewer: ```suggestion
  After a full Sidewalk Bluetooth LE deinitialization and reinitialization cycle, advertising may fail to restart.
  When a Bluetooth LE connection is established, the Bluetooth stack stops advertising, but the internal advertising state may remain stale.
```

>   On the next ``sid_start()``, ``sid_ble_advert_update()`` can call the Bluetooth HCI API with an invalid advertising set, which results in ``Advertising failed to update (err -22)`` and ``sid_start()`` returning ``SID_ERROR_GENERIC``.

Reviewer: ```suggestion
  On the next ``sid_start()`` call, ``sid_ble_advert_update()`` might call the Bluetooth HCI API with an invalid advertising set.
  This results in an ``Advertising failed to update (err -22)`` error and causes ``sid_start()`` to return ``SID_ERROR_GENERIC``.
```

  **Affected platforms:** All platforms.

>   **Workaround:** Reset the device after ``sid deinit`` before calling ``sid init`` again, or avoid the deinit and reinit cycle without a full device reset.

Reviewer: ```suggestion
  **Workaround:** Reset the device after ``sid deinit`` and before calling ``sid init`` again. 
  Alternatively, avoid the deinit and reinit cycle unless you perform a full device reset.
```
